### PR TITLE
Change makefile to make using libopaque as part of other libraries easier

### PR DESCRIPTION
**Pull request by WorldofJARcraft**
# Summary
Changes:
- added libopaque.a target
- made CFLAGS overwritable

# Reasoning

For one of my projects, I include libopaque into a shared library.
In order to make distribution of the library I am working on easier, I wanted to link libopaque statically. Thus, I added a libopaque.a as target in the makefile. Now I can include libopaque.a in my shared library and I only need to distribute one library.

Also, I have noticed that some of the default CFLAGS in the makefile might be problematic in two different ways. 
First, older gccs like 7.5 do not understand some of the flags, e.g. `-fstack-clash-protection` and `-fcf-protection=full`. 
Also, when distributing the library to other systems, since `-march=native` is used, the compiler might have used instructions that are not available on the other system, leading to SIGILL crashes.
Thus, I have made the CFLAGS overwritable, so that problematic flags can be disabled by the library user.

### CODE REVIEW

File: src/makefile [hunk 1]
```diff
@@ -1,10 +1,13 @@
 PREFIX?=/usr/local
 LIBS=-lsodium
 DEFINES=
-CFLAGS=-march=native -Wall -O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2 -fasynchronous-unwind-tables -fpic -fstack-clash-protection -fcf-protection=full -Werror=format-security -Werror=implicit-function-declaration -Wl,-z,defs -Wl,-z,relro -ftrapv -Wl,-z,noexecstack $(DEFINES)
+CFLAGS?=-march=native -Wall -O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2 -fasynchronous-unwind-tables -fpic -fstack-clash-protection -fcf-protection=full -Werror=format-security -Werror=implicit-function-declaration -Wl,-z,defs -Wl,-z,relro -ftrapv -Wl,-z,noexecstack $(DEFINES)
 LDFLAGS=-g $(LIBS)
 CC=gcc
 SOEXT=so
+AEXT=a
```
Comment (stef) [inline src/makefile:8]: what would be the aext on windows?

**Comment by stef** [comment]
sounds all reasonable. good work, useful contribution! 

one question, in what project are you using this?

thank you very much!

File: src/makefile (hunk 1 above)
Comment (WorldofJARcraft) [reply src/makefile:8]: In the Visual Studio toolchain, it would be .lib.
However, in the gcc toolchain (e.g. in MinGW), it is also .a.

**Comment by stef** [comment]
delicious!

File: src/makefile (hunk 1 above)
Comment (stef) [reply src/makefile:8]: ah, right. should've remembered .lib

**Pull request merged**